components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -387,7 +428,28 @@ private OAuthConsumerAppDTO createOAuthApp(ApplicationRegistrationRequest regist
         oAuthConsumerApp.setTokenType(registrationRequest.getTokenType());
         oAuthConsumerApp.setBackChannelLogoutUrl(
                 validateBackchannelLogoutURI(registrationRequest.getBackchannelLogoutUri()));
+        if (CollectionUtils.isNotEmpty(registrationRequest.getAudiences())) {
+            OAuth2ServiceComponentHolder.setAudienceEnabled(true);
```

Comment:
Why do we have to set this?

Reply:
This needs to be set to true if there are any audience values for a service provider. there is also a checkbox in management console to reflect this. 
![image](https://user-images.githubusercontent.com/36619815/114132210-8e948280-9921-11eb-8f61-ef8bd8617b85.png)

Reply:
I don't think we need to set this here.
This would probably be a check at startup to see if relevant tables are available to store audiences. (Anyways this is not needed now)

@emswbandara any ideas why we still have this check?

Reply:
yes we can get rid of this check now. Shall we create a seperate git issue to track it?